src/util/permissions.spec.ts

```diff
@@ -110,16 +112,48 @@ describe("General util functions for permissions feature", () => {
       },
     ];
 
+    const metadata: LxdConfigOptions = {
+      configs: {
+        cluster: {},
+        instance: {},
+        "network-bridge": {},
+        "network-macvlan": {},
+        "network-ovn": {},
+        "network-physical": {},
+        "network-sriov": {},
+        project: {},
+        server: {},
+        "storage-btrfs": {},
+        "storage-ceph": {},
+        "storage-cephfs": {},
+        "storage-cephobject": {},
+        "storage-dir": {},
+        "storage-lvm": {},
+        "storage-powerflex": {},
+        "storage-zfs": {},
+      },
+      entities: {
+        server: [
+          {
+            name: "admin",
+            description: "admin entitlement description",
+          },
+        ],
+      },
+    };
+
     const entitlementOptions = generateEntitlementOptions(
       resourceType,
       permissions,
+      metadata,
```

Comment:
Should we keep / add one test case for backwards compatibility, where metadata is null?

Reply:
done :+1: